tree-sitter-dtd/grammar.js

```diff
@@ -330,19 +334,16 @@ module.exports = grammar({
 
     EncName: _ => /[A-Za-z][A-Za-z0-9._\-]*/,
 
-    PI: $ => prec.left(seq(
+    PI: $ => seq(
       '<?',
-      // FIXME: disallow /[Xx][Mm][Ll]/
-      alias($.Name, $.PITarget),
-      // FIXME: disallow '?>'
-      O(seq($._S, /.*/)),
+      alias($._non_xml_name, $.PITarget),
+      O(seq($._S, /(\?[^>]|[^?]*)/)),
```

Comment:
I tried this myself but it doesn't work. 

```javascript
// expected: '?' | actual: '' 
/(\?[^>]|[^?]*)/.exec('?')[0]
```

It probably needs a scanner like this:

https://github.com/ObserverOfTime/tree-sitter-xml/blob/a3acc6f8ee7bf120d62fc056ec0b6d9af0781155/tree-sitter-xml/src/scanner.c#L14-L30